desktop/core/src/desktop/kt_renewer.py

```diff
@@ -50,15 +50,15 @@ def renew_from_kt():
       subp_stdout = subp.stdout.readlines()
       subp_stderr = subp.stderr.readlines()
 
-      subp_stdout = [line.decode() for line in subp.stdout.readlines()]
-      subp_stderr = [line.decode() for line in subp.stderr.readlines()]
+      subp_stdout = [line.decode() for line in subp_stdout]
+      subp_stderr = [line.decode() for line in subp_stderr]
 
       LOG.error("Couldn't reinit from keytab! `kinit' exited with %s.\n%s\n%s" % (
                 subp.returncode, "\n".join(subp_stdout), "\n".join(subp_stderr)))
       if retries >= max_retries:
         LOG.error("FATAL: max_retries of %s reached. Exiting..." % max_retries)
-        sys.exit(subp.returncode)
```

Comment:
This will looping forever after removal. Without exit. Eventually let the KT RENEWER process exit, I think it won't bob up the issue in CM.
@amitsrivastava please correct me if I am wrong.

Reply:
Yes, I too thought the same, if there is an actual failure and not an intermittent issue, this will loop forever. I'll use the previous "sys.exit(subp.returncode)" itself.